assets/js/event-providers/woocommerce.js

```diff
@@ -19,13 +19,164 @@
 		return;
 	}
 
+	const {
+		currency: globalCurrency,
+		products: globalProducts,
+		purchase,
+		add_to_cart: addToCart,
+	} = global._googlesitekit?.wcdata;
+
+	if ( addToCart ) {
+		const { price } = addToCart;
+
+		const eventData = formatEventData( price, globalCurrency, addToCart );
+
+		global._googlesitekit?.gtagEvent?.( 'add_to_cart', eventData );
+	}
+
+	if ( purchase ) {
+		const { id, totals, items } = purchase;
+
+		const eventData = formatEventData(
+			totals.total_price,
+			totals.currency_code,
+			items,
+			id,
+			totals.shipping_total,
+			totals.tax_total
+		);
+
+		global._googlesitekit?.gtagEvent?.( 'purchase', eventData );
+	}
+
 	const body = jQuery( 'body' );
 
-	body.on( 'added_to_cart', () => {
-		global._googlesitekit?.gtagEvent?.( 'add_to_cart' );
+	// eslint-disable-next-line camelcase
+	body.on( 'added_to_cart', ( event, fragments, cart_hash, $button ) => {
+		const productID = parseInt( $button.data( 'product_id' ), 10 );
+		const productData =
+			globalProducts?.find( ( product ) => product?.id === productID ) ||
+			{};
+		const { price } = productData;
+
+		const eventData = formatEventData( price, globalCurrency, productData );
+		global._googlesitekit?.gtagEvent?.( 'add_to_cart', eventData );
 	} );
 
-	body.on( 'checkout_place_order_success', () => {
-		global._googlesitekit?.gtagEvent?.( 'purchase' );
+	jQuery(
+		'.products-block-post-template .product, .wc-block-product-template .product'
+	).each( function () {
+		const productCard = jQuery( this );
```

Comment:
Usually, when something is wrapped with jQuery, then the `$` prefix is used. So for this variable its name should be `$productCard`.